File: api/Woodshed.Application/Models/Response/Posts/PostResponse.cs
namespace Woodshed.Application.Models.Response.Posts;

public class PostResponse
{
    public required string Id { get; set; }
    public required string Content { get; set; }
    public DateTime CreatedAt { get; set; }
    public required string UserId { get; set; }
    public required int CommentsCount { get; set; }
    public required int LikesCount { get; set; }
}

File: client/src/features/posts/schemas/response/PostResponse.ts
import { UserAccountResponseSchema } from "@/shared/schemas/response/UserAccountResponse";
import { z } from "zod";

export const PostResponseSchema = z.object({
  id: z.string(),
  content: z.string(),
  hasBeenEdited: z.boolean(),
  createdAt: z.iso.datetime(),
  editedAt: z.iso.datetime().nullable(),
  userId: z.string(),
  commentsCount: z.number(),
  likesCount: z.number(),
  user: UserAccountResponseSchema,
});

export type PostResponse = z.infer<typeof PostResponseSchema>;